air/src/options.rs

```diff
@@ -0,0 +1,168 @@
+    pub fn new(
+        num_queries: usize,
+        blowup_factor: usize,
+        grinding_factor: u32,
+        field_extension: FieldExtension,
+        fri_folding_factor: usize,
+        fri_remainder_max_degree: usize,
+        hash_fn: HashFunction,
+    ) -> Self {
+        let proof_options = WinterProofOptions::new(
+            num_queries,
+            blowup_factor,
+            grinding_factor,
+            field_extension,
+            fri_folding_factor,
+            fri_remainder_max_degree,
+        );
+        let exec_options = ExecutionOptions::default();
+        Self {
+            exec_options,
+            proof_options,
+            hash_fn,
+        }
+    }
+
+    /// Creates a new preset instance of [ProvingOptions] targeting 96-bit security level.
+    ///
+    /// If `recursive` flag is set to true, proofs will be generated using an arithmetization-
+    /// friendly hash function (RPO). Such proofs are well-suited for recursive proof verification,
+    /// but may take significantly longer to generate.
+    pub fn with_96_bit_security(recursive: bool) -> Self {
+        if recursive {
+            let proof_options = WinterProofOptions::new(27, 8, 16, FieldExtension::Quadratic, 4, 7);
+            Self {
+                exec_options: ExecutionOptions::default(),
+                proof_options,
+                hash_fn: HashFunction::Rpo256,
+            }
+        } else {
+            let proof_options =
+                WinterProofOptions::new(27, 8, 16, FieldExtension::Quadratic, 8, 255);
+            Self {
+                exec_options: ExecutionOptions::default(),
+                proof_options,
+                hash_fn: HashFunction::Blake3_192,
+            }
+        }
+    }
+
+    /// Creates a new preset instance of [ProvingOptions] targeting 128-bit security level.
+    ///
+    /// If `recursive` flag is set to true, proofs will be generated using an arithmetization-
+    /// friendly hash function (RPO). Such proofs are well-suited for recursive proof verification,
+    /// but may take significantly longer to generate.
+    pub fn with_128_bit_security(recursive: bool) -> Self {
+        if recursive {
+            let proof_options = WinterProofOptions::new(27, 16, 21, FieldExtension::Cubic, 4, 7);
+            Self {
+                exec_options: ExecutionOptions::default(),
+                proof_options,
+                hash_fn: HashFunction::Rpo256,
+            }
+        } else {
+            let proof_options = WinterProofOptions::new(27, 16, 21, FieldExtension::Cubic, 8, 255);
+            Self {
+                exec_options: ExecutionOptions::default(),
+                proof_options,
+                hash_fn: HashFunction::Blake3_256,
+            }
+        }
+    }
+
+    /// Sets [ExecutionOptions] for this [ProvingOptions].
+    /// This sets the maximum number of cycles a program is allowed to execute as well as
```

Comment:
nit: we should add a blank line between the intro sentence and the rest of the comment. Otherwise, doc formatter will not format them well.